enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java

```diff
@@ -332,27 +337,27 @@ public AuthenticationContext authenticate(RequestContext requestContext) throws
                         String subPolicyName = authenticationContext.getTier();
                         requestContext.addMetadataToMap("ratelimit:subscription", subscriptionId);
                         requestContext.addMetadataToMap("ratelimit:usage-policy", subPolicyName);
-                        if (datastore.getSubscriptionPolicyByName(subPolicyName) != null &&
-                                StringUtils.isNotEmpty(orgList)) {
-                            SubscriptionPolicy subPolicy = datastore.getSubscriptionPolicyByName(subPolicyName);
-                            Set<String> orgSet = Stream.of(orgList.trim().split("\\s*,\\s*"))
-                                    .collect(Collectors.toSet());
-                            if (StringUtils.isNotEmpty(subPolicy.getOrganization()) &&
-                                    orgSet.contains(subPolicy.getOrganization()) || orgList.equals("*")) {
-                                requestContext.addMetadataToMap("ratelimit:organization", subPolicy.getOrganization());
-                            } else {
-                                requestContext.addMetadataToMap("ratelimit:organization",
-                                        APIConstants.SUPER_TENANT_DOMAIN_NAME);
-                            }
+
+                        String matchedApiOrganizationId = requestContext.getMatchedAPI().getOrganizationId();
+                        if (datastore.getSubscriptionPolicyByOrgIdAndName(matchedApiOrganizationId, subPolicyName)
+                                != null) {
+                            SubscriptionPolicy subPolicy = datastore.getSubscriptionPolicyByOrgIdAndName
+                                    (matchedApiOrganizationId, subPolicyName);
+                            String metaDataOrgId = StringUtils.isNotEmpty(subPolicy.getOrganization()) &&
```

Comment:
We can remove this check StringUtils.isNotEmpty(subPolicy.getOrganization()), if we update the getOrganization() to return orgID or carbon.super as suggested in https://github.com/wso2/product-microgateway/pull/3512/files#r1601739840